src/main/features/certificateOfService/routes/what-documents.ts

```diff
@@ -0,0 +1,33 @@
+import * as express from 'express'
+import { Paths } from 'certificateOfService/paths'
+
+import { Form } from 'app/forms/form'
+import { FormValidator } from 'app/forms/validation/formValidator'
+import { WhatDocuments } from 'app/forms/models/whatDocuments'
+import ErrorHandling from 'common/errorHandling'
+import { DraftService } from '../../../services/draftService'
+
+function renderView (form: Form<WhatDocuments>, res: express.Response): void {
+  res.render(Paths.whatDocumentsPage.associatedView, { form: form })
+}
+
+export default express.Router()
+  .get(Paths.whatDocumentsPage.uri, (req: express.Request, res: express.Response) => {
+    renderView(new Form(res.locals.user.legalCertificateOfServiceDraft.document.whatDocuments), res)
+  })
+  .post(Paths.whatDocumentsPage.uri, FormValidator.requestHandler(WhatDocuments, WhatDocuments.fromObject),
+    ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+      const form: Form<WhatDocuments> = req.body
+      if (form.hasErrors()) {
+        renderView(form, res)
+      } else {
+        res.locals.user.legalCertificateOfServiceDraft.document.whatDocuments = form.model
+        res.locals.user.legalUploadDocumentDraft.document.fileToUpload = undefined
+        await new DraftService().save(res.locals.user.legalCertificateOfServiceDraft, res.locals.user.bearerToken)
+        await new DraftService().save(res.locals.user.legalUploadDocumentDraft, res.locals.user.bearerToken)
```

Comment:
I think this would be clearer if you grouped the res.locals part and the save together,
```
res.locals.user.legalCertificateOfServiceDraft.document.whatDocuments = form.model
await new DraftService().save(res.locals.user.legalCertificateOfServiceDraft, res.locals.user.bearerToken)
```

also can you extract res.locals.user out to a variable so that there is type information please?

Reply:
done